### DISCUSSION TITLE
Dynamic aggregation without a server

### DISCUSSION THREAD

**Discussion by AhmedThahir** [predict-idlab/plotly-resampler#100] [Q&A] [answered]
I like to make a quick draft reports by exporting Jupyter Notebook to HTML, before making dash applications.

This is what I could make of from the documentation.
![image](https://user-images.githubusercontent.com/94914628/178223203-5d2d9e20-acbf-4d4f-a545-38a8466ff98d.png)

**Is it possible for me to achieve dynamic aggregation without a server?**

**Comment by jonasvdd** [comment, marked as answer]
Hi @AhmedThahir,

Great question, however for me, it is not 100% clear what you mean with *dynamic aggregation without a server*. Do you mean:
* Allowing dynamic aggregation within a serialized HTML of a notebook: **this will never work**, as our dynamic aggregation logic is written in Python, and thus a python kernel is needed to achieve this.
* Allowing dynamic aggregation without starting a server: this can be achieved with the `FigureWidgetResampler`. 

Additionally, *as for now*, both `FigureWidget` and `FigureWidgetResampler` do not serialize into a plotly-interactive figure when converting a notebook into HTML. As a workaround, you can use the `.show` method on such a figure. I suppose this should be included aswell into the FAQ section.

Hope this answers your question.

**Comment by AhmedThahir** [comment]
Yeppp exactly what I thought. Thanks again! 

Again, this can be found from the documentation, but it's not very clear. If you could add this also as 
a question in the FAQ, it'd be great.

**Comment by AhmedThahir** [comment]
> ... as our dynamic aggregation logic is written in Python, and thus a python kernel is needed to achieve this.

Not for now, but a potential idea 
to consider would be to use PyScript only for the dynamic section? I'm not sure if it'd work tho.

**Reply by jvdd** [reply]
Very interesting remark! Will create an issue indicating that we should further explore this option :)
